### PR TITLE
Add a padding generator handler

### PR DESCRIPTION
**Pull request by jcague**
**Description**

As part of Simulcast/SVC it adds a new handler to generate padding. It will be useful when we need to increase bitrates and sender bandwidth estimations. 

- [x] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

Not needed.

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  }
+
+  auto processor = pipeline->getService<RtcpProcessor>();
+  if (processor) {
+    max_video_bw_ = processor->getMaxVideoBW();
+  }
+}
+
+void RtpPaddingGeneratorHandler::read(Context *ctx, std::shared_ptr<dataPacket> packet) {
+  RtpUtils::forEachRRBlock(packet, [this](RtcpHeader *chead) {
```
Comment (lodoyun) [inline erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp]: I'm thinking we should move as much as we can to the stats. Here it makes sense to react as soon as we get the info but I keep thinking we should avoid parsing the same packets n times.

File: erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  RtcpHeader *chead = reinterpret_cast<RtcpHeader*>(packet->data);
+  bool is_higher_sequence_number = false;
+  if (packet->type == VIDEO_PACKET && !chead->isRtcp()) {
+    is_higher_sequence_number = isHigherSequenceNumber(packet);
+    if (!first_packet_received_) {
+      started_at_ = clock_->now();
+    }
+    first_packet_received_ = true;
+    if (fast_start_ && clock_->now() - started_at_ > kFastStartMaxDuration) {
+      ELOG_WARN("Fast start disabled");
```
Comment (lodoyun) [inline erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp]: See log level comment above

File: erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+void RtpPaddingGeneratorHandler::read(Context *ctx, std::shared_ptr<dataPacket> packet) {
+  RtpUtils::forEachRRBlock(packet, [this](RtcpHeader *chead) {
+    if (chead->packettype == RTCP_PS_Feedback_PT && chead->getBlockCount() == RTCP_AFB) {
+      char *uniqueId = reinterpret_cast<char*>(&chead->report.rembPacket.uniqueid);
+      if (!strncmp(uniqueId, "REMB", 4)) {
+        remb_value_ = chead->getREMBBitRate();
+      }
+    } else if (fast_start_ && chead->packettype == RTCP_Receiver_PT && chead->getSourceSSRC() == audio_source_ssrc_) {
+      if (chead->getFractionLost() > kMaxFractionLostAllowed) {
+        ELOG_WARN("Fast start disabled");
```
Comment (lodoyun) [inline erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp]: Probably DEBUG or INFO would be better here

File: erizo_controller/erizoAgent/log4cxx.properties [hunk 4]
```diff
@@ -59,3 +59,4 @@ log4j.logger.rtp.SenderBandwidthEstimationHandler=WARN
 log4j.logger.rtp.StatsCalculator=WARN
 log4j.logger.rtp.LayerDetectorHandler=WARN
 log4j.logger.rtp.PliPacerHandler=WARN
+log4j.logger.rtp.RtpPaddingGeneratorHandler=WARN
```
Comment (lodoyun) [inline erizo_controller/erizoAgent/log4cxx.properties]: Can you please also add this to the tests log4cxx.properties?

File: erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp (hunk 1 above)
Comment (jcague) [inline erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp]: :+1:

File: erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp (hunk 3 above)
Comment (jcague) [inline erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp]: good catch!

File: erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp (hunk 2 above)
Comment (jcague) [inline erizo/src/erizo/rtp/RtpPaddingGeneratorHandler.cpp]: good catch! I used those while testing it manually

File: erizo_controller/erizoAgent/log4cxx.properties (hunk 4 above)
Comment (jcague) [inline erizo_controller/erizoAgent/log4cxx.properties]: sure!

**Pull request merged**